plugins/gradle/gradle-plugin/src/main/resources/META-INF/thorntail-scripts/configure-tasks.groovy

```diff
@@ -0,0 +1,29 @@
+import org.gradle.api.tasks.testing.Test
+
+/*
+ * Copyright 2018 Red Hat, Inc, and individual contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+// Given our current integration model, we end up in a scenario where we are unable to perform all integrations from
+// Java layer as that would mean that we now need to have a dependency on a newer version of the Gradle plugin.
```

Comment:
Hmm... I briefly worked with Gradle around their 1.0 time, and I think that even back then, their Java API already exposed enough to find all tasks of type `Test` in the project. In fact, I think the Groovy scripts use the same API that you can use from Java. So I'm not quite sure that we need to add an extra Groovy script; but of course you have much more experience with Gradle than me :-)

Reply:
So the problem that I faced was the following,

```
Test (task) --(references)-- DefaultJavaForkOptions
DefaultJavaForkOptions --(extends)--> DefaultProcessForkOptions --(implements)--> ProcessForkOptions
```

Now, `ProcessForkOptions` isn't available on the classpath, which causes the compilation to fail (as it is an internal API). Since Gradle doesn't ship an API compatible jar that we can use, we end up having to resort to a runtime resolution instead of relying at Reflection at compile time. Between the two choices, the runtime option keeps things simple. I was planning on removing all the reflection code in favor of clear concise Groovy code. Hence the decision to go with this approach. But I am definitely open to suggestions.

Reply:
Understood, thanks!